### PR TITLE
Remove meta classes

### PR DESCRIPTION
**Pull request by p-se**
This PR will remove unnecessary meta classes and introduce a global Ceph MGR instance as replacement for the former injection of the Ceph MGR instace into classes.

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/HACKING.rst [hunk 1]
```diff
@@ -315,21 +315,21 @@ Now only authenticated users will be able to "ping" your controller.
...
-Each controller class derived from ``BaseController``has a class property that
-points to the manager module global instance. The property is named ``mgr``.
-There is another class property called ``logger`` to easily add log messages.
+We provide the manager module instance as a global variable that can be
+imported in any module. We also provide a logger instance in the same way.
 
 Example::
 
   import cherrypy
+  from .. import logger, mgr_inst as mgr
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/HACKING.rst]: can we either use `from .. import mgr` or `from .. import mgr_inst`?

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/tools.py [hunk 2]
```diff
@@ -84,39 +82,16 @@ def json_error_page(status, message, traceback, version):
...
-        cls._mgr_module = value
-
-
-class BaseController(six.with_metaclass(BaseControllerMeta, object)):
+class BaseController(object):
     """
     Base class for all controllers providing API endpoints.
     """
-    _mgr_module = None
+    mgr = None  # type: Module
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/tools.py]: remove this line

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/HACKING.rst (hunk 1 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/HACKING.rst]: I agree, I'll change it to just `mgr`

**Review by sebastian-philipp — DISMISSED** [review]

**Comment by sebastian-philipp** [comment]
This is actually a bugfix: 
```
Error EINVAL: Traceback (most recent call last):
  File "/home/sebastian/Repos/ceph/src/pybind/mgr/dashboard_v2/module.py", line 212, in handle_command
    Auth.set_login_credentials(cmd['username'], cmd['password'])
  File "/home/sebastian/Repos/ceph/src/pybind/mgr/dashboard_v2/controllers/auth.py", line 89, in set_login_credentials
    Auth.mgr.set_config('username', username)
AttributeError: 'NoneType' object has no attribute 'set_config'
```

**Review by rjfd — DISMISSED** [review]
lgtm

**Review by sebastian-philipp — DISMISSED** [review]
lgtm

**Review by LenzGr — DISMISSED** [review]

**Review by sebastian-philipp — APPROVED** [review]
Argh. please keep the PR as it is. I don't want to review it again and again :-P

**Review by s0nea — APPROVED** [review]

**Pull request merged**